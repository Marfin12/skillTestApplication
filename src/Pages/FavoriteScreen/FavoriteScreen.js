import * as React from 'react';
import {View, Text} from 'react-native';

const FavoriteScreen = (): React.Node => {
  return (
    <View>
      <Text>Favorite Screen</Text>
    </View>
  );
};

export default FavoriteScreen;
