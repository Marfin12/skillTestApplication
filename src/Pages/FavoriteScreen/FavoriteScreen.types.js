import * as React from 'react';
import { View } from 'react-native';

import TextField from '../../Components/TextField';

import styles from './FavoriteScreen.styles';

const FavoriteScreen = () => (
  <View style={styles.container}>
    <TextField> Favorite Screen </TextField>
  </View>
);

export default FavoriteScreen;
