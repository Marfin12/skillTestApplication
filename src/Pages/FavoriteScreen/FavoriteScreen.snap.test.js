import React from 'react';
import renderer from 'react-test-renderer';
import FavoriteScreen from './FavoriteScreen';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode),
  };
});
jest.mock('../../Components/TextField', () => 'TextField');

test('renders favorite screen light mode', () => {
  const ScreenWrapper = renderer
    .create(<FavoriteScreen />)
    .toJSON();
  expect(ScreenWrapper).toMatchSnapshot();
});

test('renders favorite screen dark mode', () => {
  const ScreenWrapper = renderer
    .create(<FavoriteScreen />)
    .toJSON();
  expect(ScreenWrapper).toMatchSnapshot();
});
