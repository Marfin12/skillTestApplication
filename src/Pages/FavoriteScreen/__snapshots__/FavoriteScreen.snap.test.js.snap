// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders favorite screen dark mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextField>
     Favorite Screen 
  </TextField>
</View>
`;

exports[`renders favorite screen light mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextField>
     Favorite Screen 
  </TextField>
</View>
`;
